src/ContentBundle/Controller/Backend/ContentController.php

```diff
@@ -274,12 +274,16 @@ public function duplicateAction($id, $content)
      */
     public function editAction(Request $request, $id)
     {
-        $this->denyAccessUnlessGranted('CONTENT_EDIT');
-
         /** @var ContentManager $manager */
         $manager = $this->get('opifer.content.content_manager');
         $em = $manager->getEntityManager();
         $content = $manager->getRepository()->find($id);
+
+        $accessAttributes['permission'] = 'CONTENT_EDIT';
+        $accessAttributes['content_roles'] = $content->getRoles();
+
+        $this->denyAccessUnlessGranted($accessAttributes);
```

Comment:
This should be simplified to `$this->denyAccessUnlessGranted('CONTENT_EDIT', $content);`